LibTest/async/Stream/Stream.error_A01_t02.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion @Since("2.5")
+///   Stream<T>.error( Object error, [ StackTrace? stackTrace ])
+///
+/// Creates a stream which emits a single error event before completing.
+///
+/// This stream emits a single error event of `error` and `stackTrace` and then
+/// completes with a done event.
+///
+/// @description Checks that if `stackTrace` is specified.
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() async {
+  StackTrace stackTrace = StackTrace.fromString("My StackTrace");
+  Stream stream = Stream.error("Stream.error", stackTrace);
+  try {
+    var v = await stream.first;
+    Expect.fail("Unexpected event $v");
+  } catch (e, st) {
+    Expect.equals(stackTrace, st);
```

Comment:
Can't assume that stack traces preserve identity.
I'd do `Expect.equals("My StackTrace", st.toString());`.

(It's *probably* preserved here, but it isn't necessarily by `Error.throwWithStackTrace` because that goes through the platform throw. And we don't *promise* to not use `Error.throwWithStackTrace` internally in `Stream.error`.)